### PR TITLE
Zijie lint fix for reports component Phase 3

### PR DESCRIPTION
**Pull request by iTvX**
# Description
Lint fix for Reports component, this is phase 3.

## Files changed
WbsTable directory

- WbsTable/WbsTable.jsx 
- WbsTable/index.js

# How to Test

To test this pull request:

- Check out the current branch
- Run `npm install` and `npm run start:local` to run this PR locally  
- Log in as an admin user
- Go to Reports,  Check functionality is the same as before. (Please record a video)
- Check the functionality of the report WbsTable is normal. (Please record a video)
- Check all the functionality of the report is the same as before, and see if any bugs have appeared. (Please record a video)

Optional:
- Set up ESLint in VSCode/ IntelliJ IDEA, I personally use IntelliJ IDEA
- Delete .eslintignore file in the root: `rm .eslintignore`
- `npx eslint src/components/Reports/WbsTable/WbsTable.jsx`
- `npx eslint src/components/Reports/WbsTable/index.js`
- `git restore .`

I have complete confidence that **the code will pass the ESLint tests.** Therefore, I kindly request that you **focus on testing the frontend**. Specifically, please ensure that the **report functionality remains consistent** with how it operated previously. Thank you for your time and effort!

Screenshot:
CleanShot 2023-09-30 at 13 11 18@2x

### CODE REVIEW

**Review by DavidC0126 — COMMENTED** [review]
Hi I tested the PR and it works as intended. Good job!

eslint_3

**Comment by olenadanykh** [comment]
Hi @iTvX,

I'm not sure why, but I'm encountering a compilation error when attempting to run this branch locally. I've double-checked the current branch, performed an npm install, and then ran npm run start:local, but I'm getting the following error:



For npx eslint I am not getting any errors:

**Comment by emunkh404** [comment]
I followed the provided guide for testing, and while doing so, I encountered a few errors. However, I made necessary adjustments to the import and export statements, resulting in the expected functionality. It now works as intended. Thank you.
https://www.dropbox.com/home/Week%201%20(1)?preview=2023-09-30-1246-26.mp4
https://www.dropbox.com/home/Week%201%20(1)?preview=Screenshot+2023-09-30+125205.png
https://www.dropbox.com/home/Week%201%20(1)?preview=Screenshot+2023-09-30+125229.png
https://www.dropbox.com/home/Week%201%20(1)?preview=Screenshot+2023-09-30+125249.png

**Comment by iTvX** [comment]
Hi @olenadanykh, thank you for your detailed testing of this PR. You were correct; there was a bug. I have now fixed it. I really appreciate that!

**Review by vikrambadhan — COMMENTED** [review]
Hi, I followed your instructions and encountered these error while testing your PR

Week5 - PR13 (#1356)

**Review by TuanDinh140194 — COMMENTED** [review]
Hi! I have followed your instruction and get the below error. Maybe it has the wrong path for module components. 
Screenshot 2023-09-30 120742

**Comment by iTvX** [comment]
Hi @TuanDinh140194,

Thank you for conducting the initial tests. I apologize for the lack of clarity in the previous instructions. Here are the revised steps:

To lint the relevant files, kindly execute the following commands:
`npx eslint src/components/Reports/WbsTable/WbsTable.jsx`
`npx eslint src/components/Reports/WbsTable/index.js`

This is a fundamental test, but it's an important one, and I'm confident it will pass. I would greatly appreciate it if everyone could also run additional tests on the frontend to ensure there are no bugs.

Could you please confirm that the WbsTable report's functionality is operating as expected? Additionally, it would be helpful if you could ensure that all other functionalities in the report are working as they did previously and identify any new bugs that may have appeared.

Thank you for your time and effort.

**Comment by iTvX** [comment]
HI @TuanDinh140194 , I apologize for any confusion my previous instructions may have caused. If you have the time, could you please proceed with testing based on the latest instructions?

**Review by Haoj1 — COMMENTED** [review]
@iTvX I have tested and there is no error on my end. Good job!

**Comment by luisarevalo21** [comment]
Hello @iTvX,

I tested your PR and works as intended. Great work!

Deleting the file, running the npx eslint showed no errors. 
Screenshot 2023-10-01 at 9 02 26 PM


However, when opening Wbstable.jsx, I noticed a warning in the code. 
Screenshot 2023-10-01 at 9 05 10 PM

**Review by Sssky-18 — DISMISSED** [review]
Hi @iTvX, based on your explanation, I tested this PR and the lint tests were all passed without any error. For the report page, I didn't find any abnormal UI or buttons, so I approved this PR.  

PR#1356

**Review by eduhorta — COMMENTED** [review]
Hey @iTvX!
Everything seems to be working as expected.

image

I would to ask why did you not remove those files/folders from the `.eslintignore` and `.prettierignore`?

**Comment by iTvX** [comment]
> eduhorta

Because if you delete .eslintignore and .prettierignore, you can not pass `npm test`.

**Comment by iTvX** [comment]
> [eduhorta](/eduhorta)

Hi @eduhorta , if you delete `.eslintignore` and `.prettierignore`, you will not pass the `npm test`.

**Comment by eduhorta** [comment]
Hey @iTvX!
> Hi @eduhorta , if you delete `.eslintignore` and `.prettierignore`, you will not pass the `npm test`.

I meant why did you not remove those paths from the `.eslintignore` and `.prettierignore`?

Screenshot 2023-10-04 at 13 22 49

**Review by sxiong5 — DISMISSED** [review]
Hey @iTvX, I have tested this PR, and got the same error as @TuanDinh140194, but I checked the file, and it seems ok. So I approved it.
And maybe consider removing `src/components/Reports/**` from `.eslintignore` instead of deleting the whole file.

Screenshot 2023-10-05 at 16 53 51

**Review by Shubhu0 — APPROVED** [review]
Hi @iTvX I tested your PR and it is working as intended. Good Job! 

image

**Comment by iTvX** [comment]
> Hey @iTvX!
> 
> > Hi @eduhorta , if you delete `.eslintignore` and `.prettierignore`, you will not pass the `npm test`.
> 
> I meant why did you not remove those paths from the `.eslintignore` and `.prettierignore`?
> 
> Screenshot 2023-10-04 at 13 22 49

Hi @eduhorta ,Since the report folder contains many files, and in each phase, I only modify a specific component, I don't want to change the path. Once the entire phase is approved, we can delete that path.

**Comment by iTvX** [comment]
> Since the report folder contains many files, and in each phase, I only modify a specific component, I don't want to change the path. Once the entire phase is approved, we can delete that path.

Hi @sxiong5 , Given that the report folder has numerous files and I adjust only a specific component in each phase, removing the path would prevent other lint fixes from passing. Once the whole phase receives approval, we can eliminate that path. For testing, I recommend deleting the entire file and then using `git restore .` to restore the file.

**Comment by iTvX** [comment]
> Hey @iTvX, I have tested this PR, and got the same error as @TuanDinh140194, but I checked the file, and it seems ok. So I approved it. And maybe consider removing `src/components/Reports/**` from `.eslintignore` instead of deleting the whole file.
> 
> Screenshot 2023-10-05 at 16 53 51

Hi @sxiong5 , Given that the report folder has numerous files and I adjust only a specific component in each phase, removing the path would prevent other lint fixes from passing. Once the whole phase receives approval, we can eliminate that path. For testing, I recommend deleting the entire file and then using git restore . to restore the file.

**Review by cvtqx — APPROVED** [review]
Tested the frontend against the frontend on dev and everything seems to be functioning the same way:
https://www.loom.com/share/0078bb43464247049e0cc5b2b2c57dad?sid=9487e6cf-6a49-4e23-af19-ddf4d4c6cd8b

And I am not seeing any lint errors in the console:
 
Screen Shot 2023-10-13 at 11 54 21 AM

Great job!

**Review by yi-lin-1234 — APPROVED** [review]
Hello, functionality of the WbsTable is the same as before. And, the eslint fix worked as expected. Great work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92185695/e9d26533-fac9-4f83-adf9-1d6a41f97d83


Screenshot 2023-10-22 at 4 16 25 PM

File: src/components/Reports/WbsTable/index.js [hunk 1]
```diff
@@ -1 +1,2 @@
-export { WbsTable } from './WbsTable';
+// eslint-disable-next-line import/prefer-default-export
```
Comment (EvianTan) [inline src/components/Reports/WbsTable/index.js:1]: Please fix this error, not using "eslint-disable-next-line"

File: src/components/Reports/WbsTable/WbsTable.jsx [hunk 2]
```diff
@@ -1,23 +1,24 @@
 import { Stub } from 'components/common/Stub';
+// eslint-disable-next-line no-unused-vars
```
Comment (EvianTan) [inline src/components/Reports/WbsTable/WbsTable.jsx:2]: If there is "no-unused-vars", just delete it. You should rarely use "eslint-disable-next-line".

File: src/components/Reports/WbsTable/index.js (hunk 1 above)
Comment (iTvX) [reply src/components/Reports/WbsTable/index.js]: fixed

File: src/components/Reports/WbsTable/WbsTable.jsx (hunk 2 above)
Comment (iTvX) [reply src/components/Reports/WbsTable/WbsTable.jsx]: fixed

**Comment by EvianTan** [comment]
I just tried to build the app with the latest backend and this branch, but hit this error. Please try to resolve it.
image

File: src/components/Reports/WbsTable/WbsTable.jsx [hunk 3]
```diff
@@ -39,4 +40,4 @@ export const WbsTable = ({ wbs, skip, take }) => {
       <div>{WbsList.length > 0 ? WbsList : <Stub />}</div>
     </div>
   );
-};
+}
```
Comment (EvianTan) [inline src/components/Reports/WbsTable/WbsTable.jsx:43]: To fix the above lint issue, you'll also need to export `WbsTable` from this file. Please take this PR as an example - https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1215/files#diff-ac8689ba4b8e425618c02cb7d51230facb13753ef18963a66b2905722f6eefae

**Pull request closed without merging**